File: client/src/renderer/services/api/domains/emailConfigs.ts
/**
 * Email Configs API Client
 * Matches FastAPI router: /email-configs
 */

import { BaseApiClient } from '../base/apiClient';

// Types matching the FastAPI backend models exactly

export interface EmailFilterRule {
  field: string;
  operation: string;
  value: string;
  case_sensitive: boolean;
}

export interface EmailConfigBase {
  name: string;
  description?: string;
  email_address: string;
  folder_name: string;
  filter_rules: EmailFilterRule[];
  poll_interval_seconds: number;
  max_backlog_hours: number;
  error_retry_attempts: number;
}

export interface EmailConfig extends EmailConfigBase {
  id: number;
  is_active: boolean;
  is_running: boolean;
  emails_processed: number;
  pdfs_found: number;
  last_error_message?: string;
  last_error_at?: string;
  created_at: string;
  updated_at: string;
  last_used_at?: string;
  // Progress tracking fields
  activated_at?: string;
  last_check_time?: string;
  total_emails_processed: number;
  total_pdfs_found: number;
}

export interface EmailConfigCreate extends EmailConfigBase {}

export interface EmailConfigUpdate {
  name?: string;
  description?: string;
  email_address?: string;
  folder_name?: string;
  filter_rules?: EmailFilterRule[];
  poll_interval_seconds?: number;
  max_backlog_hours?: number;
  error_retry_attempts?: number;
}

export interface EmailAccount {
  email_address: string;
  display_name: string;
  account_type: string;
  is_default: boolean;
  provider_specific_id?: string;
}

export interface EmailFolder {
  name: string;
  full_path: string;
  message_count: number;
  unread_count: number;
  folder_type?: string;
  parent_folder?: string;
}

export interface ServiceStatusResponse {
  service: string;
  status: 'up' | 'down';
  message: string;
}

export interface ActivationResponse {
  message: string;
  config: EmailConfig;
}

/**
 * Email Configuration API client
 * Prefix: /email-configs
 */
export class EmailConfigsApiClient extends BaseApiClient {
  /**
   * Create new email configuration
   * POST /email-configs/
   */
  async createConfig(config: EmailConfigCreate): Promise<EmailConfig> {
    return this.post<EmailConfig>('/email-configs/', config);
  }

  /**
   * List all email configurations
   * GET /email-configs/
   */
  async getConfigs(): Promise<EmailConfig[]> {
    return this.get<EmailConfig[]>('/email-configs/');
  }

  /**
   * Get specific email configuration by ID
   * GET /email-configs/{config_id}
   */
  async getConfig(configId: number): Promise<EmailConfig> {
    return this.get<EmailConfig>(`/email-configs/${configId}`);
  }

  /**
   * Update email configuration
   * PUT /email-configs/{config_id}
   */
  async updateConfig(configId: number, update: EmailConfigUpdate): Promise<EmailConfig> {
    return this.put<EmailConfig>(`/email-configs/${configId}`, update);
  }

  /**
   * Activate or deactivate email configuration
   * PATCH /email-configs/{config_id}/activate?activate={true|false}
   */
  async toggleActivation(configId: number, activate: boolean): Promise<ActivationResponse> {
    return this.patch<ActivationResponse>(`/email-configs/${configId}/activate?activate=${activate}`);
  }

  /**
   * Activate email configuration
   * PATCH /email-configs/{config_id}/activate?activate=true
   */
  async activateConfig(configId: number): Promise<ActivationResponse> {
    return this.toggleActivation(configId, true);
  }

  /**
   * Deactivate email configuration
   * PATCH /email-configs/{config_id}/activate?activate=false
   */
  async deactivateConfig(configId: number): Promise<ActivationResponse> {
    return this.toggleActivation(configId, false);
  }

  /**
   * Discover all email accounts available in Outlook
   * GET /email-configs/discovery/accounts
   */
  async discoverAccounts(): Promise<EmailAccount[]> {
    return this.get<EmailAccount[]>('/email-configs/discovery/accounts');
  }

  /**
   * Discover folders for specific email account
   * GET /email-configs/discovery/folders?email_address={email}
   */
  async discoverFolders(emailAddress: string): Promise<EmailFolder[]> {
    return this.get<EmailFolder[]>(`/email-configs/discovery/folders?email_address=${encodeURIComponent(emailAddress)}`);
  }

  /**
   * Get email configs service status
   * GET /email-configs/status
   */
  async getServiceStatus(): Promise<ServiceStatusResponse> {
    return this.get<ServiceStatusResponse>('/email-configs/status');
  }
}

File: server-new/src/shared/types/email_configs.py
from dataclasses import dataclass, field
from datetime import datetime


@dataclass(frozen=True)
class FilterRule:
    field: str
    operation: str
    value: str
    case_sensitive: bool


@dataclass(frozen=True)
class EmailConfig:
    """
    Full email configuration (database record).
    Used by services and repositories for complete config data.
    """
    id: int
    name: str
    description: str | None
    email_address: str
    folder_name: str
    provider_type: str
    filter_rules: list[FilterRule]
    poll_interval_seconds: int
    max_backlog_hours: int
    error_retry_attempts: int
    is_active: bool
    activated_at: datetime | None
    last_check_time: datetime | None
    last_error_message: str | None
    last_error_at: datetime | None
    created_at: datetime
    updated_at: datetime


@dataclass(frozen=True)
class EmailConfigSummary:
    """
    Lightweight email configuration summary (for list operations).
    Used by list_configs_summary service method.
    """
    id: int
    name: str
    is_active: bool
    last_check_time: datetime | None


@dataclass(frozen=True)
class EmailConfigCreate:
    """
    Data for creating a new email configuration.
    Used by create_config service method.
    """
    name: str
    email_address: str
    folder_name: str
    description: str | None = None
    provider_type: str = "outlook_com"
    filter_rules: list[FilterRule] = field(default_factory=list)
    poll_interval_seconds: int = 5
    max_backlog_hours: int = 24
    error_retry_attempts: int = 3


@dataclass(frozen=True)
class EmailConfigUpdate:
    """
    Data for updating an email configuration.
    All fields are optional - only provided fields are updated.
    Used by update_config service method.
    """
    description: str | None = None
    filter_rules: list[FilterRule] | None = None
    poll_interval_seconds: int | None = None
    max_backlog_hours: int | None = None
    error_retry_attempts: int | None = None
    is_active: bool | None = None
    activated_at: datetime | None = None
    last_check_time: datetime | None = None
    last_error_message: str | None = None
    last_error_at: datetime | None = None